src/wazuh_db/wdb_upgrade.c

```diff
@@ -303,6 +310,19 @@ int wdb_adjust_upgrade(wdb_t *wdb, int upgrade_step) {
     }
 }
 
+int wdb_adjust_global_upgrade(wdb_t *wdb, int upgrade_step) {
+    switch (upgrade_step) {
+        case 3:
+            return wdb_global_adjust_v4(wdb);
+        default:
+            return 0;
+    }
+}
+
+int wdb_global_adjust_v4(wdb_t *wdb) {
```

Comment:
I think this logic will not be necessary.
During the upgrade we will need to have a task in wm_database to migrate all the possible unsynced groups and agent files (agent-groups, client.keys) to the global database.
During the insertion of all the info in the files **set_agent_groups_belongs()** will be called and should update this column.
Even more, it will not be necessary to migrate any of the old global.db content as it must be re-inserted by wm_database in that job.

Reply:
As we discussed with the team, we'll calculate the new `groups_hash` column values right after adding it to avoid the lack of information immediately after the upgrade. If we still have new information in the old agent-groups files, it will be inserted and properly updated also later.